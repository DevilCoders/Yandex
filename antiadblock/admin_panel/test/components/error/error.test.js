import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

import Error from 'app/components/error/error';

describe('error component', () => {
    test('without props', () => {
        const wrapper = shallow(<Error />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('with status and message', () => {
        const wrapper = shallow(<Error status={500} messageI18N='default' />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('with status and untranslated message', () => {
        const wrapper = shallow(<Error status={404} messageI18N='not-found' />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
