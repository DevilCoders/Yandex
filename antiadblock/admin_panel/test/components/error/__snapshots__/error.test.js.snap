// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`error component with status and message 1`] = `
<div
  className="error"
>
  <Connect(Header) />
  <div
    className="error__content"
  >
    <div
      className="error__status"
    >
      500
    </div>
    <div
      className="error__message"
    >
      Что-то пошло не так.
Попробуйте ещё раз или обратитесь в службу поддержки.
    </div>
  </div>
  <footer
    region="ru"
    tld="ru"
  >
    <footer__column
      key="column1"
    >
      <footer__link
        url="https://yandex.ru/company/"
      >
        О Компании
      </footer__link>
      <footer__link
        url="/support"
      >
        Помощь
      </footer__link>
    </footer__column>
    <footer__column
      key="column2"
      side="right"
    >
      <copyright
        contentRegion="ru"
      >
        <footer__link
          url="https://yandex.ru"
        >
          ООО «Яндекс»
        </footer__link>
      </copyright>
    </footer__column>
  </footer>
</div>
`;

exports[`error component with status and untranslated message 1`] = `
<div
  className="error"
>
  <Connect(Header) />
  <div
    className="error__content"
  >
    <div
      className="error__status"
    >
      404
    </div>
    <div
      className="error__message"
    >
      Возможно нужная вам страница была удалена.
Если вы считаете, что страницы нет по нашей вине, напишите нам.
    </div>
  </div>
  <footer
    region="ru"
    tld="ru"
  >
    <footer__column
      key="column1"
    >
      <footer__link
        url="https://yandex.ru/company/"
      >
        О Компании
      </footer__link>
      <footer__link
        url="/support"
      >
        Помощь
      </footer__link>
    </footer__column>
    <footer__column
      key="column2"
      side="right"
    >
      <copyright
        contentRegion="ru"
      >
        <footer__link
          url="https://yandex.ru"
        >
          ООО «Яндекс»
        </footer__link>
      </copyright>
    </footer__column>
  </footer>
</div>
`;

exports[`error component without props 1`] = `
<div
  className="error"
>
  <Connect(Header) />
  <div
    className="error__content"
  >
    <div
      className="error__status"
    >
      500
    </div>
    <div
      className="error__message"
    >
      none
    </div>
  </div>
  <footer
    region="ru"
    tld="ru"
  >
    <footer__column
      key="column1"
    >
      <footer__link
        url="https://yandex.ru/company/"
      >
        О Компании
      </footer__link>
      <footer__link
        url="/support"
      >
        Помощь
      </footer__link>
    </footer__column>
    <footer__column
      key="column2"
      side="right"
    >
      <copyright
        contentRegion="ru"
      >
        <footer__link
          url="https://yandex.ru"
        >
          ООО «Яндекс»
        </footer__link>
      </copyright>
    </footer__column>
  </footer>
</div>
`;
